modules/javafx.controls/src/test/java/test/javafx/scene/control/ComboBoxTest.java

```diff
@@ -2058,4 +2060,48 @@ public void test_rt_40012_selectedItemNotificationOnDisjointRemovesAbove() {
 
         assertEquals("ComboBox skinProperty changed more than once, which is not expected.", 1, skinChangedCount);
     }
+
+    @Test public void test_JDK_8129123() {
+        final int LIST_SIZE = 50;
+
+        ComboBox<String> comboBox = new ComboBox<>();
+
+        BorderPane p = new BorderPane();
+        p.setCenter(comboBox);
+
+        Scene scene = new Scene(p);
+
+        Stage stage = new Stage();
+        stage.setScene(scene);
+        stage.setWidth(500);
+        stage.setHeight(400);
+        stage.show();
+
+        Toolkit.getToolkit().firePulse();
+
+        for (int i = 0; i < LIST_SIZE; i++) {
+            comboBox.getItems().add(String.valueOf(i));
+        }
+        comboBox.show();
+
+        VirtualFlow<IndexedCell<?>> virtualFlow = (VirtualFlow<IndexedCell<?>>) VirtualFlowTestUtils.getVirtualFlow(comboBox);
+
+        int index = 0;
+        comboBox.getSelectionModel().select(index);
+        Toolkit.getToolkit().firePulse();
+
+        int first = virtualFlow.getFirstVisibleCell().getIndex();
+        int last = virtualFlow.getLastVisibleCell().getIndex();
+        assertTrue(" visible range [" + first + ", " + last + "] must include " + index,
+                first <= index  && index <= last);
+
```

Comment:
You have tested first and last index selection.
It will be nicer to have a case where we select index = LIST_SIZE/2

Reply:
Agreed, mid point test added